rust/ffi/node/src/lib.rs

```diff
@@ -82,13 +82,26 @@ fn runtime<'a, C: Context<'a>>(cx: &mut C) -> NeonResult<&'static Runtime> {
 
 fn database_new(mut cx: FunctionContext) -> JsResult<JsPromise> {
     let path = cx.argument::<JsString>(0)?.value(&mut cx);
+    let aws_creds = get_aws_creds(&mut cx, 1)?;
```

Comment:
It looks like you're getting a single argument here at index `1` but you're actually passing in 3 different arguments at index `1`, `2`, and `3`.  How does that work?

Reply:
This `get_aws_cred` get three parameters starting from position 1. 

I reused the logic. But we will get ride of it soon in favor of the new napi based API.